### PR TITLE
Feature/7 bd repository

### PR DESCRIPTION
**Pull request by WhiteSenat0r**
Created repository for the database, removed redundant usings

### CODE REVIEW

File: Infrastructure/Repository/GenericRepository.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+public class GenericRepository<TEntity> : IRepository<TEntity>
+    where TEntity : class
+{
+    protected GenericRepository(ApplicationContext dbContext) => 
+        Context = dbContext ?? throw new ArgumentNullException(nameof(dbContext));
+
+    private ApplicationContext Context { get; init; }
+
+    public async Task<IEnumerable<TEntity>> GetAllEntitiesAsync() =>
+        await Context.Set<TEntity>().ToListAsync();
```
Comment (IcticStep) [inline Infrastructure/Repository/GenericRepository.cs:16]: Move async/await state machine higher.

File: Infrastructure/Repository/GenericRepository.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    public async Task<TEntity> GetSingleEntityBySpecificationAsync(Func<TEntity, bool> predicate) =>
+        (await Context.Set<TEntity>().SingleOrDefaultAsync())!;
```
Comment (IcticStep) [inline Infrastructure/Repository/GenericRepository.cs:19]: Move async/await state machine higher.

File: Infrastructure/Repository/GenericRepository.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        await Context.SaveChangesAsync();
+    }
+    
+    public void UpdateExistingEntity(TEntity updatedEntity)
+    {
+        Context.Set<TEntity>().Update(updatedEntity);
+        Context.SaveChanges();
+    }
+
+    public virtual void RemoveExistingEntity(TEntity removedEntity)
```
Comment (IcticStep) [inline Infrastructure/Repository/GenericRepository.cs:33]: Why virtual?

File: Infrastructure/Repository/GenericRepository.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+namespace Infrastructure.Repository;
+
+public class GenericRepository<TEntity> : IRepository<TEntity>
+    where TEntity : class
+{
+    protected GenericRepository(ApplicationContext dbContext) => 
+        Context = dbContext ?? throw new ArgumentNullException(nameof(dbContext));
+
+    private ApplicationContext Context { get; init; }
```
Comment (IcticStep) [inline Infrastructure/Repository/GenericRepository.cs:13]: Change to readonly field.

**Review by IcticStep — CHANGES_REQUESTED** [review]

**Comment by WhiteSenat0r** [comment]
Fixed mentioned issues in a new commit.

File: Infrastructure/Repository/GenericRepository.cs [hunk 5]
```diff
@@ -7,32 +7,32 @@ namespace Infrastructure.Repository;
 public class GenericRepository<TEntity> : IRepository<TEntity>
     where TEntity : class
 {
+    private readonly ApplicationContext _context;
+    
     protected GenericRepository(ApplicationContext dbContext) =>
```
Comment (IcticStep) [inline Infrastructure/Repository/GenericRepository.cs:12]: Constructor should be after properties.

**Review by IcticStep — APPROVED** [review]

**Pull request merged**